From 3828a5fb3516395ea8ffd7bf58a322971180473c Mon Sep 17 00:00:00 2001
From: Thiago Macieira <thiago.macieira@intel.com>
Date: Fri, 7 Sep 2018 13:20:42 -0700
Subject: [PATCH] Add qmake config "fat-static-lto"

This is useful for Linux distributions, since installed static libraries
should not carry LTO information, but that information is useful during
Qt's own build. This feature should be used alongside some compiler-
specific method of removing the LTO information from the static
libraries prior to installation, so only the regular part remains.

For current GCC versions, this command suffices:
  strip -R '.gnu.lto*' -R '.gnu.debuglto*' libname.a

Change-Id: I495bc19409f348069f5bfffd155237ade9f4b42f
---
 mkspecs/features/ltcg.prf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mkspecs/features/ltcg.prf b/mkspecs/features/ltcg.prf
index f2dbc6fa9e..3a7a36d6ef 100644
--- a/mkspecs/features/ltcg.prf
+++ b/mkspecs/features/ltcg.prf
@@ -22,7 +22,7 @@ CONFIG(release, debug|release) {
         }
     }
 
-    fat-lto {
+    fat-lto|if(static:fat-static-lto) {
         QMAKE_CFLAGS_LTCG += $$QMAKE_CFLAGS_LTCG_FATOBJECTS
         QMAKE_CXXFLAGS_LTCG += $$QMAKE_CXXFLAGS_LTCG_FATOBJECTS
     }
-- 
2.22.0

